drivers/flash/nrf_qspi_nor.c

```diff
@@ -503,27 +505,86 @@ static inline int qspi_nor_read_id(struct device *dev,
 	return 0;
 }
 
+static inline nrfx_err_t read_non_aligned(struct device *dev, off_t addr,
+					  void *dest, size_t size)
+{
+	uint8_t __aligned(WORD_SIZE) buf[WORD_SIZE * 2];
+	uint8_t *dptr = dest;
+
+	off_t flash_prefix = WORD_SIZE - (addr % WORD_SIZE) % WORD_SIZE;
```

Comment:
You need to use parentheses here. Otherwise, the second modulo operation will be performed before the subtraction, and consequently it will be useless.

```suggestion
	off_t flash_prefix = (WORD_SIZE - (addr % WORD_SIZE)) % WORD_SIZE;
```